blog-posts/getting-started-gatsby-next-remix.md

```diff
@@ -0,0 +1,306 @@
+leadboxText: Have you learned something about Gatsby, Next.js or Remix? Then join us to learn more!
+attribution:
+    creator: Daniel Kudela
+    source: https://unsplash.com/photos/K59MPijxRh0
+---
+
+
+
+
+Before starting a React based project, the same question often arises: Which framework should be used? Of course, this question cannot be answered in a general way, since each framework has its advantages and disadvantages for specific use cases. Nevertheless, you can compare the frameworks, especially the concepts behind them.
+
+That's why we decided to develop a simple blog application in Gatsby, Next.js and Remix to compare the frameworks. This application can read data from markdown files, display them and create dynamic routes depending on them. This article will discuss the differences in the development process, as well as the basic technical concepts in the frameworks. The three versions of the comparison blog project were each deployed on Vercel or Netlify and have a public GitHub repository.
+
+
+> 💡 For simplification, style and images are not included in the examples. The detailed versions are available in the GitHub repositories for the comparison blog project.
+
+
+- **Blog app Gatsby Version:** [Preview](https://gatsby-comparison-blog.vercel.app/) and [Repository](https://github.com/hofmann-felix/gatsby-comparison-blog)
+- **Blog app Next.js Version:** [Preview](https://next-js-comparison-blog.vercel.app/) and [Repository](https://github.com/hofmann-felix/next.js-comparison-blog)
+- **Blog app Remix Version:** [Preview](https://remix-comparison-blog.netlify.app/) and [Repository](https://github.com/hofmann-felix/remix-comparison-blog)
+
+  ![blog-overview.png](images/gastby-next-remix-screenshot-blog-overview.png)
+
+
+## Setting up a project
+
+#### Gatsby
+
+To start a Gatsby project you have to run `npx gatsby new`. The setup process offers many individualization possibilities: Besides the project name and path, a CMS, a styling system (e.g. styled-components, emotion, ...), and additional features with other plugins (e.g. mdx support) can be added optionally. Gatsby has built-in out-of-the-box TypeScript support, so there is no distinction between TypeScript and JavaScript in the project setup.
+
+#### Next.js
+
+The command `npx create-next-app@latest` can additionally be extended with the flag `--ts` to add direct TypeScript support to a new Next.js project. The following setup process does not provide any customization options apart from the project name.
+
+#### Remix
+
+A Remix project can be set up with the command `npx create-remix@latest`. Then you can specify the project name, choose between different host options (e.g. Netlify or Vercel) and between TypeScript and JavaScript.
+
+## Static Site Generation
+
+With static site generation (SSG), the HTML is generated once at built time and reused for each request. Of the three frameworks in this article, only Remix does not support SSG.
```

Comment:
For Gatsby there is no running process involved (if you are not opt-in to SSR) and Next.js can't be run without the process running, so you can deploy to S3 and such. Is this correct? That would help here to explain as it's not clear what keeps running after the build.

Reply:
If you only have static pages in next you can run `next export`. Would be the same as `gatsby build`.

https://nextjs.org/docs/advanced-features/static-html-export#unsupported-features

My guess is that not a lot of projects will do this, as Next gives you so much more features with the running process.